app/src/docs/components/DocsLayout/Navigation/index.jsx

```diff
@@ -54,55 +63,69 @@ class Navigation extends React.Component<Props, State> {
-        const { subNavigationItems } = this.props
+    generateNavigation(topLevelNav, index) {
+        const navItem = docsNav[topLevelNav]
+        const isActiveSection = this.isActiveSection(navItem)
+        const { pathname } = this.props.location
 
         return (
-            <Scrollspy items={subNavigationItems && Object.keys(subNavigationItems)} currentClassName={styles.active}>
-                {Object.entries(subNavigationItems).map((subNavigationItem) => (
-                    <li key={`item-${subNavigationItem[0]}`} className={styles.navListItem}>
-                        <a
-                            href={`#${subNavigationItem[0]}`}
-                        >
-                            {String(subNavigationItem[1])}
-                        </a>
-                    </li>
-                ))}
-            </Scrollspy>
+            <React.Fragment key={index}>
+                <li className={styles.navListItem}>
+                    <Link className={isActiveSection ? styles.active : ''} to={navItem.root}>{topLevelNav}</Link>
+                </li>
+                <ul
+                    className={cx(styles.subNavList, {
+                        [styles.show]: isActiveSection,
+                        [styles.hide]: !isActiveSection,
+                    })}
+                >
+                    {/* Render subNav contents */}
+                    {Object.keys(navItem).map((subKey) => {
+                        if (subKey !== 'root') {
+                            return (
+                                <li key={subKey} className={styles.navListItem}>
+                                    <Link
+                                        className={pathname.includes(navItem[subKey]) ? styles.active : ''}
+                                        to={navItem[subKey]}
+                                    >
+                                        {subKey}
+                                    </Link>
+                                </li>)
+                        }
+                        return null
+                    })
+                    }
+                </ul>
+            </React.Fragment>
         )
     }
 
-    parseCurrentPage() {
-        const { navigationItems } = this.props
-
-        return Object.entries(navigationItems).map((navListItem) => (
-            this.props.location.pathname === navListItem[1] ?
-                (
-                    <li key={`item-${String(navListItem[1])}`} className={styles.navListItem}>
-                        <Link to={formatPath(String(navListItem[1]))}>
-                            {navListItem[0]}
-                        </Link>
-                    </li>
-                ) : null
-        ))
+    isActiveSection(subNavList) {
+        let match = false
+
+        Object.keys(subNavList).forEach((subKey) => {
+            if (this.props.location.pathname.includes(subNavList[subKey])) {
+                match = true
+            }
+        })
+
+        return match
```

Comment:
minor, but `some` stops iterating as soon as it finds a match, while `forEach` always iterates over the whole list:

```js
isActiveSection(subNavList) {
    return Object.keys(subNavList).some((subKey) => (
        this.props.location.pathname.includes(subNavList[subKey])
    ))
}
```

Reply:
this makes me very happy : ) thanks for the tip